app/views/developers/index.html.erb

```diff
@@ -1,26 +1,42 @@
 <p id="notice"><%= notice %></p>
-
 <h1>Developers</h1>
+<%= form_for @developers_search_form, url: developers_path, method: :get do |search_form| %>
+  Email : <%= search_form.text_field :email %>
+  Language (Code format) : <%= search_form.text_field :language %>
+  Programming Languages : <%= search_form.text_field :programming_language %>
+  <%= search_form.submit "Search" %>
+<% end %>
 
 <table>
   <thead>
     <tr>
       <th>Email</th>
-      <th colspan="3"></th>
+      <th>Languages</th>
+      <th>Programming Languages</th>
+      <th colspan="3">Actions</th>
     </tr>
   </thead>
 
   <tbody>
     <% @developers.each do |developer| %>
       <tr>
         <td><%= developer.email %></td>
+        <td>
+          <%= developer.languages.map{|l| l.code}.join(",") %>
```

Comment:
@truongle7793 
we can change 
`developer.languages.map{|l| l.code}.join(",")`
to
`
developer.languages.pluck(:code).join(', ')
`

Reply:
@thienhv 
Thanks
It look easier to read, i have made change.